pkg/check-depdiff/dependencies.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2022 Security Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package depdiff
+
+// ChangeType is the change type (added, updated, removed) of a dependency.
+type ChangeType string
+
+const (
+	Added   ChangeType = "added"
+	Updated ChangeType = "updated"
+	Removed ChangeType = "removed"
+)
+
+// IsValid determines if a ChangeType is valid.
+func (ct *ChangeType) IsValid() bool {
+	switch *ct {
+	case Added, Updated, Removed:
+		return true
+	default:
+		return false
+	}
+}
+
+// Dependency is a dependency.
+type Dependency struct {
+	// ChangeType indicates whether the dependency is added, updated, or removed.
+	ChangeType ChangeType `json:"change_type"`
```

Comment:
In the scorecard codebase, we use camel case for this, so let's use that here too. Example: `changeType`

Reply:
I mean the json tag, yes. We agreed that this structure should be generic and not tied to an API, right? My understanding is that this structure is the one used tby scorecard to return the results, not the structure used to query the GH API.

Let me know if I mis-understood something

Reply:
@laurentsimon  PTAL at the latest changes, thx :D